### PR TITLE
AWS example now configures a DLQ for inbox notifications

### PR DESCRIPTION
**Pull request by wimagguc**
### Public-Facing Changes

The SNS subscription now configures a dead letter queue, so messages are not dropped to the ground in case Console is unavailable.

Tested end-to-end with the latest console & data-platform. Indeed, after an object lands in the bucket, SNS will attempt to deliver the notification max `numRetries`(+1) times:

```
{
  "Type" : "Notification",
  "MessageId" : "7d70475b-5c8a-5bd4-9cbe-1ad56997a6fd",
  "TopicArn" : "arn:aws:sns:us-east-1:096363264652:org-slug-inbox-bucket-delete-me-sns-topic",
  "Subject" : "Amazon S3 Notification",
  "Message" : "{\"Records\":[{\"eventVersion\":\"2.1\",\"eventSource\":\"aws:s3\", ... \"eTag\":\"8964da5a74d6afdac379cc72575fef79\",\"sequencer\":\"00651432DCA7EF4799\"}}}]}",
...
```

...and after we run out of retries, the message appears in SQS as expected:

```
$ aws sqs receive-message --queue-url 'https://sqs.us-east-1.amazonaws.com/096363264652/org-slug-inbox-bucket-delete-me-sns-topic-sqs-dlq'

{
    "Messages": [
        {
            "Body": "{\n  \"Type\" : \"Notification\",\n  \"MessageId\" : \"7d70475b-5c8a-5bd4-9cbe-1ad56997a6fd\",\n  ...\\\"eTag\\\":\\\"8964da5a74d6afdac379cc72575fef79\\\",\\\"sequencer\\\":\\\"00651432DCA7EF4799\\\"}}}]}\",\n 
...
```

### CODE REVIEW

File: primary-site/aws/modules/s3/main.tf [hunk 1]
```diff
@@ -2,11 +2,6 @@ resource "aws_s3_bucket" "bucket" {
   bucket = var.bucket_name
 }
 
-resource "aws_s3_bucket_acl" "bucket_acl" {
-  bucket = aws_s3_bucket.bucket.id
-  acl    = "private"
-}
```
Comment (wimagguc) [inline primary-site/aws/modules/s3/main.tf:8]: Unrelated change; as of April 2023 S3 [automatically blocks public access](https://aws.amazon.com/about-aws/whats-new/2022/12/amazon-s3-automatically-enable-block-public-access-disable-access-control-lists-buckets-april-2023/), and so creating this `aws_s3_bucket_acl` resource will throw an error.

File: primary-site/aws/modules/sns/main.tf [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+}
+
 resource "aws_sns_topic" "topic" {
   name = var.topic_name
   delivery_policy = jsonencode({
     "http" : {
       "defaultHealthyRetryPolicy" : {
         "numRetries" : 100,
+        "minDelayTarget" : 20, // In seconds; default is 20
+        "maxDelayTarget" : 20, // In seconds; default is 20
```
Comment (wimagguc) [inline primary-site/aws/modules/sns/main.tf:12]: Without specifying the two new min/max delay parameters, this line would throw an "Invalid parameter: DeliveryPolicy: http.defaultHealthyRetryPolicy.minDelayTarget must be specified" error with the latest aws provider.

**Review by foxglovenes — APPROVED** [review]
LGTM. Thank you!
